java/mdsobjects/src/test/java/MDSplus/MdsTreeTest.java

```diff
@@ -44,12 +44,14 @@ public void testData(){
 		tree = new MDSplus.Tree("java_test0",-1,"READONLY");
 		Assert.assertEquals(true, tree.isReadOnly() );
 		tree.close();
+		tree.close();
```

Comment:
I think since you added a lot of tree.close the context may want to stick with the tree object and you free it in finallize, i.e. when the object is destroyed. if you call edit on an open tree it should not free but reuse it and so forth. explicitly calling close though should free it up. maybe do a quit automatically if in edit mode. this would allow it to implement the Closable interface.